lib/src/main/java/growthbook/sdk/java/multiusermode/configurations/GlobalContext.java

```diff
@@ -0,0 +1,48 @@
+package growthbook.sdk.java.multiusermode.configurations;
+
+import com.google.gson.JsonObject;
+import growthbook.sdk.java.Experiment;
+import growthbook.sdk.java.multiusermode.util.TransformationUtil;
+import lombok.Builder;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+@Data
+@Builder
+@Slf4j
+public class GlobalContext {
+
+    /**
+     * Keys are unique identifiers for the features and the values are Feature objects.
+     * Feature definitions - To be pulled from API / Cache
+     */
+    @Nullable
+    private JsonObject features; // do you need features Json? -- How is it handled in JS SDK?
+
+    @Nullable
+    private String featuresJson;
+
+    @Nullable
+    private JsonObject savedGroups;
+
+    private List<Experiment> experiments;
+
+    public JsonObject getFeatures() {
+        if (this.features == null) {
+            this.features = TransformationUtil.transformFeatures(this.featuresJson);
+            // Both are constructed then n there at the time of processing. Restructure them?
+            // TBD:M read saved Groups
+            // TBD:M experiments
+        }
+        return this.features;
+    }
+
+    public void loadEncryptedFeatures(String decryptionKey) {
+        if (this.features == null) {
+            this.features = TransformationUtil.transformEncryptedFeatures(this.featuresJson, decryptionKey);
+        }
+    }
```

Comment:
Getting / decrypting features seems like something that should be done during init, not lazily loaded.  If we did that, we wouldn't need to store `featuresJson` here at all, just the final `features` object.

Reply:
yeah, I agree! but the lombok library isn't supporting custom methods. I'll have to write custom builders to achieve this. Let me try.